Improve describe_version for git-fleximod

billsacks opened the pull request:
### Description of changes

Until recently, the describe_version script worked with manage_externals. In ca22dbf, @jedwards4b changed it to work with git-fleximod, but in doing so removed some of its features. This PR brings back its original features but changes it to work with git-fleximod.

Specific features this brings are:
- Provides CESM git hash and status for files in the CESM repository (in addition to those in sub-components, which are reported by git-fleximod)
- Allows invoking describe_version from some other path (e.g., from within `cime/scripts`, you could run `../../describe_version` and it would work the same as if you ran `cd ../../; ./describe_version`)

### Specific notes

Contributors other than yourself, if any:

Fixes ESCOMP/CESM#245

User interface changes?: No

Testing performed (automated tests and/or manual tests): Manual testing of describe_version script

jedwards4b commented on describe_version:
```diff
@@ -1 +1,69 @@
+#!/usr/bin/env python
+"""Describes the CESM version and any local modifications"""
+
+from __future__ import print_function
```
I don't think that you need this line.

billsacks replied:
Good point - this was a carry-over from the past version of this. I have removed it.

jedwards4b reviewed (changes requested):
Thanks for this.

jedwards4b commented:
@billsacks We have changed the workflow for cesm - please redirect your PR to the cesm3-alphabranch.   Thanks

billsacks commented:
> We have changed the workflow for cesm - please redirect your PR to the cesm3-alphabranch.

Thanks for the reminder. I have rebased this and changed the base for this PR.
